### ISSUE TITLE
Implement System Interrupts

### ISSUE THREAD

**Issue by roysmanfo** [roysmanfo/Flux#46]
# System Interrupts
Interrupts should be a way for the commands and the user to execute a given function once a condition is met, even
if the command isn't being run actively.

> **NOTE**: This will allow commands to add arbitrary code to the terminal, so this feature should be disabled by default and not accessible to the commands.

___
### Interrupt creation
To create a new interrupt, some information must be provided, like:

#### Condition 
The condition on which the terminal will determine whether or not to execute the interrupt

#### Lifes
How often must the function be executed (for each life the interrupt gets executed once).
If immortal, after the first execution, the interrupt won't be executed again even if the condition is met again

#### Callback
The function to execute once the condition is met
___
### Prototypes 

``` py
# returns True if the interrupt has been created and added
def create_interrupt(condition: Callable[[...], bool], callback: Callable[[...], None], exec_once: bool, ...) -> bool
```

``` py
# returns a handle to the interrupt 
def create_interrupt(condition: Callable[[...], bool], callback: Callable[[...], None], exec_once: bool, ...) -> IntHande
```
___

_This issue is for tracking the development of this feature.
All updates will be linked to this issue._

**Comment by roysmanfo** [comment]
now to create a new interrupt the following information needs to be provided (**more options may be added later, this is not final**):
name | description
---|----
event | can be one of the supported `Signals` or `EventTriggers`. Specifies when to execute the interrupt
target | the actual code to execute once the specified event occurs. Takes **at least** 2 parameters by default (`signum` and `frame`, like a normal signal)
args (optional) |  the arguments needed by the target function
kwargs (optional) |  is a dictionary of keyword arguments for the target invocation. Defaults to {}.
exec_once (default: True) | if set to False, the interrupt will be executed at each event, as long as the command is still alive

After registering a new Interrupt, the function returns an interrupt handle (IHandel) to the `Interrupt`, which will be useful when interacting with it

**Comment by roysmanfo** [comment]
the most part of this has been implemented in #54. ([here is the exact commit](../pull/54/commits/b01bb220f645dc812f2b68a4f050beb8a3275939))
